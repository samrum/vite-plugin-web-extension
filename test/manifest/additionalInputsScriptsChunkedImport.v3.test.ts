import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("additionalInputsScriptsChunkedImport");

runManifestV3Test("additionalInputsScriptsChunkedImport", () => ({}), {
  additionalInputs: {
    scripts: [`${resourceDir}/script1.js`, `${resourceDir}/script2.js`],
  },
});
