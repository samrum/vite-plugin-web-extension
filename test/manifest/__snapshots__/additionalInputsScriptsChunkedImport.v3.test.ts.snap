// Vitest Snapshot v1

exports[`additionalInputsScriptsChunkedImport - Manifest V3 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"script1\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js",
    "fileName": "assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js": {
        "code": "log(\\"script1\\");",
        "originalLength": 50,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 15,
      },
    },
    "name": "test/manifest/resources/additionalInputsScriptsChunkedImport/script1",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"script2\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js",
    "fileName": "assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js": {
        "code": "log(\\"script2\\");",
        "originalLength": 50,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 15,
      },
    },
    "name": "test/manifest/resources/additionalInputsScriptsChunkedImport/script2",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"web_accessible_resources\\": [
    {
      \\"resources\\": [
        \\"assets/log.js\\",
        \\"assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js\\",
        \\"assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js\\",
        \\"test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js\\",
        \\"test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js\\"
      ],
      \\"matches\\": [
        \\"<all_urls>\\"
      ],
      \\"use_dynamic_url\\": true
    }
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script1.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/additionalInputsScriptsChunkedImport/script2.js\\"))})();",
    "type": "asset",
  },
]
`;
